bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/ItemChannelLinkRegistry.java

```diff
@@ -9,24 +9,91 @@
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.items.ItemRegistry;
 import org.eclipse.smarthome.core.thing.ChannelUID;
 import org.eclipse.smarthome.core.thing.Thing;
 import org.eclipse.smarthome.core.thing.ThingRegistry;
 import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.UID;
 import org.eclipse.smarthome.core.thing.link.events.LinkEventFactory;
+import org.osgi.service.component.ComponentContext;
 
 /**
  * {@link ItemChannelLinkRegistry} tracks all {@link ItemChannelLinkProvider}s
  * and aggregates all {@link ItemChannelLink}s.
  *
  * @author Dennis Nobel - Initial contribution
+ * @author Markus Rathgeb - Respect item existence
  *
  */
 public class ItemChannelLinkRegistry extends AbstractLinkRegistry<ItemChannelLink> {
 
+    /*
+     * Methods of the AbstractRegistry does not respect the component life cycle.
+     * The method AbstractRegistry.addProvider executes logic if the component has not been activated, too.
+     * This could result in some wired runtime behavior because member variables that are injected by the service
+     * management could be present or not.
+     */
+    boolean active = false;
+    private final LinkedList<Provider<ItemChannelLink>> waitingProviders = new LinkedList<>();
+
     private ThingRegistry thingRegistry;
+    private ItemRegistry itemRegistry;
+
+    protected void activate(ComponentContext context) {
+        itemRegistry.addRegistryChangeListener(itemRegistryChangeListener);
+        synchronized (this) {
+            active = true;
+            while (!waitingProviders.isEmpty()) {
+                super.addProvider(waitingProviders.removeFirst());
+            }
+        }
+    }
+
+    protected void deactivate() {
+        synchronized (this) {
+            active = false;
+        }
+        itemRegistry.removeRegistryChangeListener(itemRegistryChangeListener);
+    }
+
+    private final RegistryChangeListener<Item> itemRegistryChangeListener = new RegistryChangeListener<Item>() {
+        @Override
+        public void added(Item element) {
+            final String itemName = element.getName();
+            for (final ItemChannelLink itemChannelLink : getAll()) {
+                if (itemChannelLink.getItemName().equals(itemName)) {
+                    notifyListenersAboutAddedElement(itemChannelLink, true);
+                }
+            }
+        }
+
+        @Override
+        public void removed(Item element) {
+            final String itemName = element.getName();
+            for (final ItemChannelLink itemChannelLink : getAll()) {
+                if (itemChannelLink.getItemName().equals(itemName)) {
+                    notifyListenersAboutRemovedElement(itemChannelLink, true);
+                }
+            }
+        }
+
+        @Override
+        public void updated(Item oldElement, Item element) {
+            if (!oldElement.equals(element)) {
+                this.removed(oldElement);
```

Comment:
"updated" here is called if the item meta data changes - its category, label or whatever. It is NOT called for state changes, so this should be completely independent here.